Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/FunctionBlock.java

```diff
@@ -0,0 +1,248 @@
+    @FXML private BorderPane outputSpace;
+
+    /** The space containing all the arguments of the function. */
+    private ArgumentSpace argumentSpace;
+    
+    /** The space in which to nest the FunctionBlock's inner parts. */
+    @FXML private Pane nestSpace;
+    
+    /** The space in which the information of the function is displayed. */
+    @FXML private Pane functionInfo;
+    
+    /**
+     * Method that creates a newInstance of this class along with it's visual
+     * representation
+     *
+     * @param name
+     *            The name of the function.
+     * @param type
+     *            The function's type (usually a FuncT).
+     * @param pane
+     *            The parent pane in which this FunctionBlock exists.
+     */
+    public FunctionBlock(String name, Type type, CustomUIPane pane) {
+        super(pane);
+        this.name = new SimpleStringProperty(name);
+
+        this.loadFXML("FunctionBlock");
+        Expr signature = new Ident(getName());
+
+        // Collect argument types
+        ArrayList<String> args = new ArrayList<>();
+        Type t = null;
+        try {
+            t = signature.getType(pane.getEnvInstance()).prune();
+        } catch (HaskellException e1) {
+            throw new FunctionDefinitionException();
+        }
+        int inputCount = 0;
+        while (t instanceof FuncT) {
+            FuncT ft = (FuncT) t;
+            args.add(ft.getArgs()[0].toHaskellType());
+            t = ft.getArgs()[1];
+            inputCount++;
+        }
+        
+        argumentSpace = new ArgumentSpace(this, inputCount);
+        argumentSpace.setKnotIndex(getAllInputs().size());
+        
+        nestSpace.getChildren().add(argumentSpace);
+        argumentSpace.knotIndexProperty().addListener(e -> invalidateKnotIndex());
+        
+        // Create an anchor for the result
+        output = new OutputAnchor(this);
+        outputSpace.setCenter(output);
+        
+        // Make sure the prefWidth is correctly updated.
+        this.prefWidthProperty().bind(functionInfo.widthProperty().add(argumentSpace.prefWidthProperty()));
+        
+        
+        this.setMinSize(Region.USE_PREF_SIZE, Region.USE_PREF_SIZE);
+        this.setMaxSize(Region.USE_PREF_SIZE, Region.USE_PREF_SIZE);
+        functionInfo.setMinWidth(Region.USE_PREF_SIZE);
+        functionInfo.setMaxWidth(Region.USE_PREF_SIZE);
+        
+        // Since at this point of the width of the Labels is unknown, we have to ask for another layout pass.
+        Platform.runLater(this::updateLayout);
+    }
+    
+    /** Updates the layout, if this Pane has a parent. */
+    public void updateLayout() {
+        if (this.getParent() != null) {
+            this.getParent().requestLayout();
+        }
+    }
+    
+    /** @return the name of this FunctionBlock. */
+    public final String getName() {
+        return name.get();
+    }
+
+    /** Sets the name of this FunctionBlock. */
+    public void setName(String name) {
```

Comment:
Do we want to support changing the name after construction? I think it would only lead to confusion.